### DISCUSSION TITLE
CODE RUN UNDEF

### DISCUSSION THREAD

**Discussion by akaday** [community/community#141495] [Discover: GitHub Best Practices]
Infinite Loops: Code that runs indefinitely, consuming CPU resources.

Indeed, infinite loops can be a real drag on system resources. Here’s a simple example:

javascript

Copier
while(true) {
    // This code runs indefinitely
    console.log("This will keep running!");
}
This kind of loop will continuously log to the console without stopping, effectively consuming CPU resources until the browser crashes or the process is terminated. While educational, they’re definitely not something you want running unchecked!

Ever wondered about other creative but positive coding tricks?

**Comment by GabrielAlmeidag** [comment]
@akaday 
Here are some positive coding tricks that can enhance your programming practices. First, debouncing and throttling are valuable techniques for managing event handling; debouncing limits how often a function can execute after a burst of events, while throttling ensures a function runs at a maximum frequency. Another useful trick is asynchronous programming, which utilizes promises or async/await to handle operations without blocking the main thread, thus improving performance and responsiveness. Additionally, memoization can optimize function calls by caching results of expensive computations, speeding up future calls with the same inputs. Incorporating these techniques can lead to more efficient, maintainable, and responsive code.

**Comment by novium2** [comment]
Hey @akaday and @GabrielAlmeidag We'd love to have you join discussions here, but we do require original responses, and not AI generated responses, since folks can go start a chat session with their favorite AI if they are so inclined. Thanks!
